java/micronaut-jsonview-demo-app/src/main/java/com/example/micronaut/controller/StudentController.java

```diff
@@ -75,15 +72,16 @@ public Optional<StudentView> updateStudent(Long id, @NonNull String student) {
         });
     }
 
-    @Post("/") // <8>
+    @Post// <8>
     @Status(HttpStatus.CREATED) 
-    public Optional<StudentView> create(@NonNull @Body CreateStudentDto createDto) {
-      // Use a relational operation to insert a new row in the STUDENT table
-      Student student = studentRepository.save(new Student(createDto.student(), createDto.averageGrade()));
-      // For each of the courses in createDto parameter, insert a row in the STUDENT_COURSE table
-      courseRepository.findByNameIn(createDto.courses()).stream()
-          .forEach(course -> studentCourseRepository.save(new StudentCourse(student, course)));
-      return studentViewRepository.findByStudent(student.name());
+    public StudentView create(@NonNull @Body CreateStudentDto createDto) {
+        // Must initialize data for student view by given courses
+        List<Course> courses = courseRepository.findByNameIn(createDto.courses());
+        List<StudentScheduleView> scheduleViews = courses.stream()
+                .map(c -> new StudentScheduleView(null, new CourseView(c.id(), c.name(), new TeacherView(c.teacher().id(), c.teacher().name()), c.room(), c.time()))).toList();
+        StudentView studentView = new StudentView(createDto.student(), createDto.averageGrade(),
+                scheduleViews);
```

Comment:
This returns auto increment ID created from the jdv insert and sets to the studentView instance being returned.